### ISSUE TITLE
Angular.io website is not opening in Safari browser.

### ISSUE THREAD

**Issue by Meet1202** [angular/angular#48296]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

When we open angular.io website in safari browser in mac then it giving blank page and if we do refresh that page than also it's giving blank page and when we open inspect element from developer open in safari browser and do refresh that page then it open perfectly so it's an issue I have provided screenshot.

- This when I open angular.io link in Safari browser in Mac.
![Screenshot 2022-11-30 at 6 37 28 PM](https://user-images.githubusercontent.com/41354599/204804800-e682b1c1-e8e3-4f6e-b89e-128526cbb017.png)

- This when I open inspect element of safari browser and still website is not working.
![Screenshot 2022-11-30 at 6 39 15 PM](https://user-images.githubusercontent.com/41354599/204805119-0db38b62-6f5b-4e03-a382-2f1cbb7da64d.png)

- And at last this after opening Inspect element I refresh the page then website is working properly.
![Screenshot 2022-11-30 at 6 39 27 PM](https://user-images.githubusercontent.com/41354599/204805558-5cea7fb4-a1d6-44f9-af86-bd3750bfbbeb.png)

### Please provide a link to a minimal reproduction of the bug

https://angular.io

**Comment by kbrilla** [comment]
Same hapenning on iOS 15.4

**Comment by matthewdenobrega** [comment]
I had this issue with my app - it wasn't loading on Safari 15 or in a Webview on iOS 15 (16 was fine) with a thrown exception about "Can't find variable t..." and an uninformative stack trace. 

I tracked it down to forwardRefs - I was using forwardRefs on all my ControlValuAccessors, and removing these (they are unnecessary because they are in the class decorators) resolved this issue. This was across two apps - one Ionic and one Angular material.

So it looks like there's an issue with forwardRefs, at least when defining ControlValueAccessors, in Angular 15, on Safari 15 and iOS 15 webviews.

This was not an issue on Angular14, I only started seeing these errors after I upgraded.

**Comment by devversion** [comment]
Hi all, I've tracked down the root cause. It's a very subtle bug that got fixed in Safari v16. And yes, the bug doesn't reproduce when the devtools are opened. Here is how I tracked it down:
 
1. Initially loaded the page with devtools closed, saw an error message like in https://github.com/angular/angular/issues/48296#issuecomment-1332474784. The stack did not point to anything useful- especially since source maps can hide issues.
2. Disabled source maps, closed dev-tools and re-freshed. Now the error looked like this:
![image](https://user-images.githubusercontent.com/4987015/205013485-7198c156-7254-41a7-920a-3aec23937884.png)
    * The error did not point to something useful either. Not even close to something with `$o`. Safari stack trace was incorrect.
3. Rebuilt angular.io with `optimization: false`. Fortunately the issue was reproducable even with unoptimized build (making debugging easier):
    *  ![image](https://user-images.githubusercontent.com/4987015/205013984-f6f2aaa8-cc19-475f-9b3e-05b6c27be622.png)
4. Now that the culprit JS code was finally visible- more investigation was possible.
    * The variable accessed is clearly in scope. See `notification_component_c0`.
    * There is some weird syntax for the static fields generated by the Angular compiler. e.g. notice `static #_ = this.ecmp = X`
5. I initially thought the private identifier static fields are the problem, so I tried to understand where these are coming from:
    * Looked through Angular CLI's compilation pipeline and looked for Babel plugins.
    * There is a Babel plugin generating these for older browsers not supporting `static {}` blocks.
    * Surprisingly Angular compiler does not generate `static {}` blocks- but static fields.
    * Turned out TypeScript converts static fields to static blocks if `useDefineForClassFields = false` (which is the case in Angular v15 w/ target: 2022)
    * I've tried to isolate the problem and create a minimal reproduction, but couldn't reproduce. Since the issue was already very hard to track down (given devtools **cannot being opened**)- I've still assumed something odd could go on in Safari..
    * Looked through WebKit bug tracker and commits but couldn't find anything- except for private identifier methods fix in the parser— could have been related.
    * Ultimately I concluded that static private identifier fields are not the issue.
6. I've tried to "tamper" the main bundle and modify parts to see what the issue could be, slowly removing parts of the code snippet in (3). 
    * Tracked it down to "still" happen when there is code like `(0, x).method()` in the static field initializer. In other scenarios, the error quickly was gone- almost seeming like it's non-deterministic.
    *  My assumption was that it's a scope tracking issue in the WebKit JS parsing.
    * Looked through Webkit commits and found a fix (in the reasonable time frame of Safari v15.6 -> v16)
    * Issue description seems very similar/matching: https://github.com/WebKit/WebKit/commit/e8788a34b3d5f5b4edd7ff6450b80936bff396f2
    * https://bugs.webkit.org/show_bug.cgi?id=236843

TL;DR: Angular emits valid JavaScript that is subject to a bug in Safari that got fixed in Safari v16. The issue does not reproduce when devtools are opened- and the Safari bug is related to scope tracking in the WebKit JS parsing. https://github.com/WebKit/WebKit/commit/e8788a34b3d5f5b4edd7ff6450b80936bff396f2.

We will look into downleveling static fields for Safari <16 so that Angular apps remain compatible with that browser, if selected via browserslist (or the default one).
